internal/plugin/grpc_client_plugin.go

```diff
@@ -46,7 +49,10 @@ func NewGrpcClient(agentConfig *config.Config) *GrpcClient {
 
 		return &GrpcClient{
 			config:          agentConfig,
+			isConnected:     false,
+			instances:       []*v1.Instance{},
```

Comment:
It seems a little bit weird to me that a gRPC client would have a list of instances. Would it be better if the status update was handled by the instance plugin instead of our gRPC client?

Reply:
If the gRPC client plugin is disabled then the instances plugin wont know that and try to make a request.
The instance plugin should only be concerned with finding instances and nothing else.

The only reason I need to store them is because the Process function is getting called before the Init function which might need further investigation to see if there is a way to prevent this.

Reply:
Yeah not calling `Init()` before `Process()` sounds like a problem alright. Do you wanna look at it in another PR or do it here?